shell/src/main/java/alluxio/cli/fs/command/DuCommand.java

```diff
@@ -108,28 +130,50 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
    * @param statuses the statuses of files and folders
    * @param readable whether to print info of human readable format
    * @param summarize whether to display the aggregate summary lengths
+   * @param groupByWorker whether to display In-Alluxio groupByWorker
    * @param addMemory whether to display the memory size and percentage information
    */
   protected static void getSizeInfo(AlluxioURI path, List<URIStatus> statuses,
-      boolean readable, boolean summarize, boolean addMemory) {
+      boolean readable, boolean summarize, boolean groupByWorker, boolean addMemory) {
+    Optional<String> workerHostName = groupByWorker ? Optional.of("total") : Optional.empty();
     if (summarize) {
       long totalSize = 0;
       long sizeInAlluxio = 0;
       long sizeInMem = 0;
+      Map<String, Long> distributionMap = new HashMap<>();
       for (URIStatus status : statuses) {
         if (!status.isFolder()) {
           long size = status.getLength();
           totalSize += size;
           sizeInMem += size * status.getInMemoryPercentage();
           sizeInAlluxio += size * status.getInAlluxioPercentage();
         }
+        if (groupByWorker) {
+          fillDistributionMap(distributionMap, status);
+        }
       }
       String sizeMessage = readable ? FormatUtils.getSizeFromBytes(totalSize)
           : String.valueOf(totalSize);
       String inAlluxioMessage = getFormattedValues(readable, sizeInAlluxio / 100, totalSize);
-      String inMemMessage = addMemory
-          ? getFormattedValues(readable, sizeInMem / 100, totalSize) : "";
-      printInfo(sizeMessage, inAlluxioMessage, inMemMessage, path.toString());
+      Optional<String> inMemMessage = addMemory
+          ? Optional.of(getFormattedValues(readable, sizeInMem / 100, totalSize))
+              : Optional.empty();
+
+      printInfo(sizeMessage, inAlluxioMessage, inMemMessage, path.toString(), workerHostName);
+
+      //If inMemMessage is present, underline information composed with In-Alluxio size
+      //and worker host name need a "" as placeholder. If inMemMessage is not present,
+      //which means inMemMessage is not exist, underline information need an empty Optional.
+      //e.g. inMemMessage is present, inMem should be ""
+      //File Size     In Alluxio       In Memory        Worker Host Name          Path
+      //2             2                2                total                     /
+      //              2                                 node1
+      //e.g. inMemMessage is not present, inMem should be an empty Optional
+      //File Size     In Alluxio       Worker Host Name          Path
+      //2             2                total                     /
+      //              2                node1
```

Comment:
```suggestion
      // If inMemMessage is present, the “In Alluxio” and "Worker Host Name"
      // columns need an empty string as placeholders. 
      // Otherwise we use an empty Optional.
      // e.g. inMemMessage is present, inMem should be ""
      // File Size     In Alluxio       In Memory        Worker Host Name          Path
      // 2                 2                    2                        total                                  /
      //                    2                                              node1
      //e.g. inMemMessage is not present, inMem should be an empty Optional
      //File Size     In Alluxio       Worker Host Name          Path
      //2                 2                    total                                  /
      //                   2                    node1
```